theano/sandbox/gpuarray/dnn.py

```diff
@@ -1472,6 +1475,27 @@ def local_log_softmax_dnn(node):
         return [new_softmax(softmax_node.inputs[0])]
 
 
+@register_opt('cudnn')
+@op_lifter([LogSoftmax])
+def local_logsoftmax_to_dnn(node, ctx_name):
+    if not dnn_available(ctx_name) or version() < 3:
+        # No log-softmax before cudnn v3
+        return
+
+    if (isinstance(node.op, LogSoftmax) and node.inputs[0].owner and
+            isinstance(node.inputs[0].owner.op, HostFromGpu)):
+
+        # Transform the input in the format expected by GpuDnnSoftmax
+        inp = node.inputs[0].owner.inputs[0]
```

Comment:
use inp=node.inputs[0]

then you won't need to tag for context as this will be done by the op_lifter.